### PR TITLE
Implement compress action for commandline use

### PR DESCRIPTION
**Pull request by bahaokten**
Fixes the issue: #5290 
Added compress commandline action to allow conversion from mmp to mmpz
Usage: `lmms --compress infile.mmp > outfile.mmpz`

### CODE REVIEW

**Comment by zonkmachine** [comment]
Thank you for working on this!

I've tested it and it works as described.

Some thoughts.

>  Usage: `lmms --compress infile.mmp > outfile.mmpz`

If you type the command wrong or try the command without all arguments in order to try out how it works you'll end up dumping binary data to the terminal and this doesn't look nice. I don't know if you can have any other problems but the purely cosmetic ones. Maybe the output could be directly dumped in a file with the same name as the input file but with the .mmpz suffix, if such a file does not already exist. In the similar way as the **--export** command works (though **--export** doesn't test if the file already exist...).

File: src/core/main.cpp [hunk 1]
```diff
@@ -161,6 +161,7 @@ void printHelp()
 		"Actions:\n"
 		"  <no action> [options...] [<project>]  Start LMMS in normal GUI mode\n"
 		"  dump <in>                             Dump XML of compressed file <in>\n"
+		"  compress <in>                         Compress file <n>\n"
```
Comment (zonkmachine) [inline src/core/main.cpp]: I think this needs improving. `compress <in> > <out>` is more like an ordinary use case. I don't know if that's good syntax too but you get the idea. Also I don't know what it looks like on non linux/unix systems.

`Compress file <n>\n"`

Did you mean `Compress file <in>\n"`

File: src/core/main.cpp [hunk 2]
```diff
@@ -427,6 +428,22 @@ int main( int argc, char * * argv )
 
 			return EXIT_SUCCESS;
 		}
+		else if( arg == "compress" || arg == "--compress")
```
Comment (zonkmachine) [inline src/core/main.cpp]:
```suggestion
		else if( arg == "compress" || arg == "--compress" )
```

File: src/core/main.cpp [hunk 3]
```diff
@@ -427,6 +428,22 @@ int main( int argc, char * * argv )
...
+		{
+			++i;
+
+			if( i == argc )
+			{
+				return noInputFileError();
+			}
+			
+			QFile f( QString::fromLocal8Bit( argv[i] ) );
+			f.open( QIODevice::ReadOnly);
```
Comment (zonkmachine) [inline src/core/main.cpp]:
```suggestion
			f.open( QIODevice::ReadOnly );
```

**Review by zonkmachine — CHANGES_REQUESTED** [review]
Just some style nitpicking.

**Comment by zonkmachine** [comment]
@bahaokten You have one more unanswered question here: https://github.com/LMMS/lmms/pull/5341#discussion_r357937386

Shouldn't it be `Compress file <in>\n"`?

File: src/core/main.cpp (hunk 1 above)
Comment (Veratil) [inline src/core/main.cpp:164]:
```suggestion
		"  compress <in>                         Compress file <in>\n"
```

File: src/core/main.cpp [hunk 4]
```diff
@@ -427,6 +428,22 @@ int main( int argc, char * * argv )
 
 			return EXIT_SUCCESS;
 		}
+		else if( arg == "compress" || arg == "--compress" )
```
Comment (Veratil) [inline src/core/main.cpp:431]: Can you update the code formatting to the [Coding Conventions](https://github.com/LMMS/lmms/wiki/Coding-Conventions) please?

Just some whitespace changes is all. 😄

**Review by Veratil — CHANGES_REQUESTED** [review]

File: src/core/main.cpp (hunk 1 above)
Comment (zonkmachine) [reply src/core/main.cpp]: Merged. It looks like the one who press the merge button on  a suggested change becomes the 'Author'.

File: src/core/main.cpp (hunk 4 above)
Comment (zonkmachine) [reply src/core/main.cpp:431]: But it's adopting to the surrounding code so I think it's good like this.

File: doc/bash-completion/lmms [hunk 5]
```diff
@@ -89,7 +89,7 @@ _lmms()
     pars_render=(--float --bitrate --format --interpolation)
     pars_render+=(--loop --mode --output --profile)
     pars_render+=(--samplerate --oversampling)
-    actions=(dump render rendertracks upgrade)
+    actions=(dump compress render rendertracks upgrade)
     actions_old=(-d --dump -r --render --rendertracks -u --upgrade)
```
Comment (zonkmachine) [inline doc/bash-completion/lmms:93]: The bash completion for compress works but it would also nice to have the old style --compress work.

Comment (zonkmachine) [reply doc/bash-completion/lmms:93]: Right. None of the other actions are bash completing with old style longopts so scratch that for this PR.

**Comment by zonkmachine** [comment]
What's left to do here?

I can compress projects from the command line. :+1: 
This looks good to go to me.

 @Veratil There hasn't been any response from OP since January. No need to wait any longer. If you want to fixup any formatting issues to your liking I think you should do so and then squash it.

**Review by zonkmachine — APPROVED** [review]

**Comment by zonkmachine** [comment]
@Veratil Merge? Do you want to touch up the formatting first?

**Comment by zonkmachine** [comment]
> @Veratil Merge? Do you want to touch up the formatting first?

OK. No answer, merging.

**Pull request merged**